test/util/bspNet/waits.ts

```diff
@@ -128,13 +135,34 @@ export const waitForBspStored = async (api: ApiPromise, checkQuantity?: number)
       if (checkQuantity) {
         invariant(
           matches.length === checkQuantity,
-          `Expected ${checkQuantity} extrinsics, but found ${matches.length} for fileSystem.bspVolunteer`
+          `Expected ${checkQuantity} extrinsics, but found ${matches.length} for fileSystem.bspConfirmStoring`
         );
       }
       const { events } = await sealBlock(api);
       assertEventPresent(api, "fileSystem", "BspConfirmedStoring", events);
       break;
     } catch {
+
+      if ( bspId ) {
+        try {
+          // In cases the bsp is submitting a proof at the same time is trying to confirmStoring
+          await assertExtrinsicPresent(api, {
+            module: "proofsDealer",
+            method: "submitProof",
+            checkTxPool: true,
+            timeout: 100
+          });
```

Comment:
This is working but not what we discussed. The `assertExtrinsicPresent` function doesn't return the actual extrinsic so I cannot check the bspId (also not the actual ID but the adddress).

I could add an optional address value to `assertExtrinsicPresent` to verify it is indeed the one that submitted the proof but it would be pretty specific to `submitProof` method.

For now this is working just to check when we have one BSP trying to store.

Reply:
@TDemeco the BSP ID in the transaction will likely be `None`, because the BSP is sending it with it's address that is linked to his BSP ID. So she would have to get the address that is the signer of the extrinsic, and compare that.